synapse/storage/databases/main/appservice.py

```diff
@@ -434,16 +437,12 @@ async def set_appservice_stream_type_pos(
                 % (stream_type,)
             )
 
-        def set_appservice_stream_type_pos_txn(txn):
-            stream_id_type = "%s_stream_id" % stream_type
-            txn.execute(
-                "UPDATE application_services_state SET %s = ? WHERE as_id=?"
-                % stream_id_type,
-                (pos, service.id),
-            )
+        stream_id_type = "%s_stream_id" % stream_type
 
-        await self.db_pool.runInteraction(
-            "set_appservice_stream_type_pos", set_appservice_stream_type_pos_txn
+        await self.db_pool.simple_update_one(
+            "application_services_state",
+            {"as_id": service.id},
+            {stream_id_type: pos},
```

Comment:
This transaction may still encounter a SerializationFailure since it's on REPEATABLE READ.

Reply:
It won't, as we use `autocommit` mode for one off queries like this, which acts more like `READ COMMITTED`

Reply:
(Which is non-obvious unless you know about it!)